Mage/src/main/java/mage/abilities/effects/common/continuous/PlayFromTopOfLibraryEffect.java

```diff
@@ -0,0 +1,99 @@
+package mage.abilities.effects.common.continuous;
+
+import mage.abilities.Ability;
+import mage.abilities.SpellAbility;
+import mage.abilities.effects.AsThoughEffectImpl;
+import mage.cards.Card;
+import mage.constants.AsThoughEffectType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.filter.FilterCard;
+import mage.game.Game;
+import mage.players.Player;
+
+import java.util.Locale;
+import java.util.UUID;
+
+/**
+ * @author nantuko, JayDi85, xenohedron
+ */
+public class PlayFromTopOfLibraryEffect extends AsThoughEffectImpl {
+
+    private final FilterCard filter;
+
+    private static final FilterCard defaultFilter = new FilterCard("play lands and cast spells");
+
+    /**
+     * You may play lands and cast spells from the top of your library
+     */
+    public PlayFromTopOfLibraryEffect() {
+        this(defaultFilter);
+    }
+
+    /**
+     * You may [play lands and/or cast spells, according to filter] from the top of your library
+     */
+    public PlayFromTopOfLibraryEffect(FilterCard filter) {
```

Comment:
For future proofing, I'd like it we had a clear split Play/Cast here, and not rely on the filter logic to not match lands.

Haven't checked all the callers yet, but there may be some that could be wrong with [[Dryad Arbor]]. Zoetic Cavern seems properly handled with The `cardToCheck = spell.getCharacteristics(game);` logic.